active/0003-extensible-resource-deployment-observability-improvements.md

```diff
@@ -0,0 +1,311 @@
+
+### Should `outputResources` include `existing` resources as well?
+
+Currently, `outputResources` only contains created or updated resources. Should we expand the property to include `existing` resources?
+
+> ✅ No, we should not change the semantics of the `outputResources` property. If there is a need for returning referenced resources in the future, a new property like `referencedResources` can be added.
+
+### [Resolved] Better name for `deploymentProviders`
+
+> ℹ️ The `extensions` property added to the deployments API response body was originally named `deploymentProviders` in the initial design.
+
+Can we come up with some better names?
+
+- `importedProviders`
+- `infraProviders`
+- `managementProviders`
+- `apiProviders`
+- `platformProviders`
+- `platforms`
+- `apis`
+- `extensions`
+- `plugins`
+- `vendors`
+- `protocols`
+- `services`
+- `workloads`
+
+> ✅ We decided to use `extensions` (or "Bicep extensions" and "ARM template extensions" for added clarity) as suggested by @anthony-c-martin for these reasons:
+> - The term `providers` is considered to be highly ambiguous, particularly within the ARM context where it typically denotes Azure resource providers, which could confuse discussions around Bicep's extensibility features. Although qualifiers can help clarify the context, finding an appropriate and precise qualifier is challenging. Additionally, the introduction of a qualifier results in the formation of a compound noun, which can be cumbersome and less user-friendly.
+> - The term `extensions` is deemed to be less prone to misunderstanding. The potential for confusion between Bicep / ARM template deployment extensions and Azure resource providers is minimal. Although the term is also used in the context of Azure extension resources, it is a less common usage. Notably, the Graph team has already adopted the term extensions over providers internally for clearer internal communication.
+> - The concept of "extensibility" has been a recurring theme in our public documentation. Adopting the term `extensions` aligns closely with this existing terminology, making it a more intuitive choice.
+> - 💪 "Bicep extension" resonates well with our established naming conventions for Bicep-related features.
+
+
+## Out of scope
+
+### Portal UI design
+
+The update to the Portal deployments blade is necessary to display the symbolic name and Bicep extension information for extensible resources. Since the portal team has ownership of the deployments blade, they hold the responsibility for both the UI design and the implementation of these changes. The addition of new properties — `extensions` in the deployments API and `extensions` in the deployment operations API — provides the Portal team with the essential data to programmatically differentiate between Azure resources and extensible resources. This capability is key to enabling the team to design a UI that effectively segregates and presents Azure resources and extensible resources.
```

Comment:
I think it's fine to consider this out of scope, as long as we're confident there's no ambiguity around whether the proposed API changes give them the right data to build the experience. Is it worth engaging to double check this (e.g. have the Portal team do a mock-up)?

Reply:
It should provide enough data to at least list the resources in the deployment details pane. However, if a clickable link to the extension resource should be provided, I'm not so certain how this would be formulated. For Azure resources, the resource ID itself is likely enough. For the AKS example, it's unclear how to link a specific AKS cluster to each individual resource in a way the portal understands. Maybe this is something the extension config REP can address. One solution could be to include a list of referenced ARM resources in the configuration detail. The portal UI could find an AKS cluster resource in that list and if it exists, associate each resource scoped to the extension to the cluster.

Reply:
Followup: Create a backlog item for clickable links for ext resources.